src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,19 +39,16 @@
     function load(templateId) {
       $scope.loadInProgress = true;
 
-      templates.getTemplate(templateId)
-        .then(function (result) {
-          $scope.template.fromName = result.from_name;
-          $scope.template.fromEmail = result.from_email;
-          $scope.template.subject = result.subject;
-          $scope.template.content = result.body;
-          $scope.template.name = result.name;
-          $scope.template.id = result.id;
-        }).catch(function (error) {
-          // TODO: do something with the error
-        }).finally(function () {
-          $scope.loadInProgress = false;
-        });
+      templates.getTemplateWithBody(templateId).then(function (result) {
```

Comment:
What do you think about return the promise?

```
return templates.getTemplateWithBody(templateId).then(function (result) {
```

Reply:
I like to be consistent with other methods like save(), they don't return promises, and the parent method is called only in a private scope at the controller.

Reply:
It is not important at this moment, I also like consistency, but I prefer moving the consistency to promises approach.

It is powerful and cheap, I mean, if you return the promise, you could take advantage of it, or simply ignore it. If you do not return the promise you _have to_ ignore it.